service/accessanalyzer/api_op_ListArchiveRules.go

```diff
@@ -113,3 +114,78 @@ func addOperationListArchiveRulesMiddlewares(stack *middleware.Stack, options Op
 	}
 	return nil
 }
+
+// ListArchiveRulesAPIClient is a client that implements the ListArchiveRules
+// operation.
+type ListArchiveRulesAPIClient interface {
+	ListArchiveRules(context.Context, *ListArchiveRulesInput, ...func(*Options)) (*ListArchiveRulesOutput, error)
+}
+
+var _ ListArchiveRulesAPIClient = (*Client)(nil)
+
+// ListArchiveRulesPaginatorOptions is the paginator options for ListArchiveRules
+type ListArchiveRulesPaginatorOptions struct {
+	// The maximum number of results to return in the request.
+	Limit *int32
+
+	// Set to true if pagination should stop if the service returns a pagination token
+	// that matches the most recent token provided to the service.
+	StopOnDuplicateToken bool
```

Comment:
It would be great to have a way to limit which operations this is generated based on if the service returns duplicate tokens or not. But i doubt there is any sane way to keep track of this and for it to be correct.

I could see this as a confusing option for users, unsure when they should set this field.  Maybe we can improve this by expanding the documentation here giving hint when to use this option.